File: test/TestingExample.ManagementApiClient/Scenario/Model/Variation.cs
using System.Globalization;

namespace TestingExample.ManagementApiClient.Scenario.Model;

public record VariationType(LocaleType Culture, SegmentType Segment);

public static class Variation
{
    public static VariationType Invariant { get; } = new(Locale.Invariant, new InvariantSegment());
    public static VariationType Culture(CultureInfo culture) => new(Locale.Culture(culture), new InvariantSegment());
    public static VariationType Culture(string culture) => Culture(CultureInfo.GetCultureInfo(culture));
}

public record LocaleType();

public record InvariantLocale() : LocaleType;
public record CultureLocale(CultureInfo Culture) : LocaleType;

public static class Locale
{
    public static LocaleType Invariant { get; } = new InvariantLocale();
    public static LocaleType Culture(CultureInfo culture) => new CultureLocale(culture);
    public static LocaleType Culture(string culture) => Culture(CultureInfo.GetCultureInfo(culture));
}


public record SegmentType();

public record InvariantSegment() : SegmentType;
public record VariantSegment(string Alias) : SegmentType;

File: test/TestingExample.Website.FunctionalTests/packages/scenario-builder/src/domain/variation.ts
export type Variation = {
  culture: Culture;
  segment: Segment;
};

export const InvariantCulture = null;

// TODO: what is the actual format for culture?
export type Culture = string | typeof InvariantCulture;

export const InvariantSegment = null;

export type Segment = string | typeof InvariantSegment;

export const Invariant: Variation = {
  culture: InvariantCulture,
  segment: InvariantSegment,
};

export function cultureVariant(culture: string): Variation {
  return {
    culture: culture,
    segment: InvariantSegment,
  };
}
